db/dcrsqlite/sqlite.go

```diff
@@ -388,8 +388,8 @@ func (db *DB) StoreBlock(bd *apitypes.BlockDataBasic, isMainchain bool, isValid
 			db.dbSummaryHeight = height
 		}
 		if db.lastStoredBlock == nil || bd.Height >= db.lastStoredBlock.Height {
-			blockdata := apitypes.BlockDataBasic(*bd)
```

Comment:
superfluous conversion since `*bd` was already a `apitypes.BlockDataBasic`.  While here, change to new and copy to do the same thing.